### PR TITLE
BoxesResponse: Fold names into top-level object

### PR DESCRIPTION
**Pull request by michaeldiamant**
Due to https://github.com/algorand/generator/issues/36, it's _not_ possible to generate working Java and Go SDK changes prior to the PR (see https://github.com/algorand/go-algorand-sdk/pull/340 as an example).

Consequently, the PR proposes re-defining the response for `/v2/applications/{application-id}/boxes` to be compatible with as-is _generator_ capabilities.

Notes:
* Since a `Box` definition exists (returning name + value), the PR proposes a new definition:  `BoxDescriptor`.  I'm open to different names.  The intent is to use a name that _may_ be extended with new fields in the future.
* Alternatives considered:
  * Keep PR as is and return array of box names rather than array of `BoxDescriptor`.  The approach is a better fit for today's model.  However, _if_ we later modify the response type, it'll introduce a backwards incompatible change.  
  * Close PR and instead modify _generator_ templates to produce expected changes.  The approach feels more error prone than the PR's approach.

### CODE REVIEW

**Review by ahangsu — COMMENTED** [review]
seems a reasonable change, I am not tagged to review so I just comment here.

It just surprise me that the generated model cannot handle response type of array of strings, and I wrote most of the code here, so I kinda feel guilty about it.

**Comment by michaeldiamant** [comment]
> I wrote most of the code here, so I kinda feel guilty about it.

@ahangsu No worries and no need to feel guilty!  It's an unexpected outcome and the SDKs are outside the scope of #4154.

**Review by algochoi — COMMENTED** [review]
Looks good to me, as discussed off line

File: daemon/algod/api/algod.oas2.json [hunk 1]
```diff
@@ -3259,10 +3273,17 @@
     "BoxesResponse": {
       "description": "Box names of an application",
```
[the remark this answers is not in the document]
Comment (michaeldiamant) [reply daemon/algod/api/algod.oas2.json:3274]: @algochoi I'm open to suggestions.  

The PR focuses on _today's_ usage under the assumption that it's _least_ likely to be misinterpreted by API users.  I imagine it's _OK_ to morph the description over time.

**Review by algochoi — APPROVED** [review]
:shipit:

**Review by algoidurovic — APPROVED** [review]
LGTM

**Pull request merged**